e2e/bridge_test.go

```diff
@@ -0,0 +1,228 @@
+          "name": "data",
+          "type": "bytes"
+        }
+      ],
+      "name": "onStateReceived",
+      "outputs": [],
+      "stateMutability": "nonpayable",
+      "type": "function"
+    }
+  ]`
+)
+
+var (
+	StateSyncerABI   = abi.MustNewABI(StateSyncerABIJSON)
+	StateRecieverABI = abi.MustNewABI(StateReceiverABIJSON)
+)
+
+func TestBridge_StateSync(t *testing.T) {
+	senderKey, senderAddr := tests.GenerateKeyAndAddr(t)
+
+	// Start root chain
+	sourceIBFT := framework.NewIBFTServersManager(
+		t,
+		IBFTMinNodes,
+		fmt.Sprintf("%ssource", IBFTDirPrefix),
+		func(i int, config *framework.TestServerConfig) {
+			config.Premine(senderAddr, framework.EthToWei(10))
+			config.SetSeal(true)
+		})
+
+	ctx1, cancel1 := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel1()
+	sourceIBFT.StartServers(ctx1)
+
+	// deploy state syncer contract
+	ctx2, cancel2 := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel2()
+	syncerContractAddr, err := sourceIBFT.GetServer(0).DeployContract(ctx2, StateSyncerByteCode, senderKey)
+	if err != nil {
+		t.Fatal(err)
+	}
+	fmt.Printf("Deployed StateSyncerByteCode %s\n", syncerContractAddr)
+
+	// setup destination chain
+	destIBFT := framework.NewIBFTServersManager(
+		t,
+		IBFTMinNodes,
+		fmt.Sprintf("%sdest", IBFTDirPrefix),
+		func(i int, config *framework.TestServerConfig) {
+			config.SetSeal(true)
+			config.Premine(senderAddr, framework.EthToWei(10))
+			config.SetUseBridge(true)
+			config.SetBridgeRootChainURL(sourceIBFT.GetServer(0).WSJSONRPCAddr())
+			config.SetBridgeRootChainContract(syncerContractAddr.String())
+			config.SetBridgeRootChainConfirmations(5)
+
+			config.SetShowsLog(i == 0)
+		})
+
+	ctx3, cancel3 := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel3()
+	destIBFT.StartServers(ctx3)
+
+	// deploy state reciever contract
+	ctx4, cancel4 := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel4()
+	reciverContractAddr, err := destIBFT.GetServer(0).DeployContract(ctx4, StateReceiverByteCode, senderKey)
+	if err != nil {
+		t.Fatal(err)
+	}
+	fmt.Printf("Deployed StateSyncerContractAddr %s\n", reciverContractAddr)
+
+	// Send state sync contract
+	ctx5, cancel5 := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel5()
+	to := types.BytesToAddress(syncerContractAddr[:])
+	input, err := StateSyncerABI.Methods["stateSync"].Encode(map[string]interface{}{
+		"contractAddress": reciverContractAddr.String(),
+		"data":            []byte{0x68, 0x65, 0x6c, 0x6c, 0x6f}, // hello
+	})
+	fmt.Printf("input %+v\n", input)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = sourceIBFT.GetServer(0).SendRawTx(ctx5, &framework.PreparedTransaction{
+		Gas:      framework.DefaultGasLimit,
+		GasPrice: big.NewInt(framework.DefaultGasPrice),
+		To:       &to,
+		From:     senderAddr,
+		Input:    input,
+	}, senderKey)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// TODO: remove of course
```

Comment:
This one made me laugh 🙂 

Remove, of course 💯 

cc @Vuksan

Reply:
Fixed b908be88021f5c3c20443ee54ad1818a979f0825